File: low-level-design/design-patterns/python/creational/abstract-factory/azure_factory.py
from factory import Factory
from azure import AzureCompute,AzureStorage,AzureNetwork


class AzureFactory(Factory):

    def createCompute(self)->AzureCompute:
        
        return AzureCompute()

    def createStorage(self)->AzureStorage:
        
        return AzureStorage()

    def createNetwork(self)->AzureNetwork:
        
        return AzureNetwork()

File: low-level-design/design-patterns/golang/creational/abstract-factory/abstract-factory/azure_factory.go
package abstractfactory

func NewAzureFactory() *AzureFactory {
	return &AzureFactory{}
}

type AzureFactory struct {
}



func (f *AzureFactory) CreateCompute() Compute {
	return NewAzureCompute()
}

func (f *AzureFactory) CreateStorage() Storage {
	return NewAzureStorage()
}

func (f *AzureFactory) CreateNetwork() Network {
	return NewAzureNetwork()
}
